R/parse_citation.R

```diff
@@ -131,27 +131,25 @@ guess_citation <- function(path) {
   # return NULL if CITATION does not exist either
   if (! citation_exists) return(NULL)
 
-  # Read DESCRIPTION to determine encoding
-  description <- desc::desc(file.path(path, "DESCRIPTION"))
+  # Read DESCRIPTION to determine meta
+  meta <- parse_package_meta(file.path(path, "DESCRIPTION"))
 
   # Read and parse CITATION
-  bib <- read_citation_with_encoding(citation_path, description$get("Encoding"))
+  bib <- read_citation_with_encoding(citation_path, meta)
```

Comment:
```suggestion
  bib <- utils::readCitationFile(citation_path, meta)
```

I am reasonably sure you can get rid of `read_citation_with_encoding()` altogether, because `readCitationFile()` should not error anymore with all meta info present, i.e. `citation(auto = meta)` should work and the error, `read_citation_with_encoding()` is trying to catch is never triggered. (I'm reasonably sure, because `readCitationFile(citfile, meta)` is what [`utils::citation()` does](readCitationFile(citfile, meta))).

There may be some use for the "get rid of the `citation(auto = meta) line" approach, though, because the CodeMeta citation field is supposed to (only) contain references to other works, not to the source code itself, IIUC.

> A citation or reference to _another_ creative work, such as another publication, web page, scholarly article, etc.
Source: https://codemeta.github.io/terms/ under "citation", emphasis mine.

So `read_citation()` could remove `citation(auto = meta)` if it is present. I wonder if this is the most efficient way to get rid of a self-citation though.

Reply:
See https://github.com/cboettig/codemeta/pull/9/commits/6aca0c78872c92e459c270908076ae6b71d26cd8